tests/lib/__init__.py

```diff
@@ -1375,3 +1375,24 @@ def __call__(
 
 
 CertFactory = Callable[[], str]
+
+# versions containing fix/backport from https://github.com/python/cpython/pull/113563
+# which changed the behavior of `urllib.parse.urlun{parse,split}`
+has_new_urlun_behavior = (
+    # https://github.com/python/cpython/commit/387ff96e95b9f8a8cc7e646523ba3175b1350669
+    (sys.version_info[:2] == (3, 12) and sys.version_info >= (3, 12, 4))
+    or
+    # https://github.com/python/cpython/commit/872000606271c52d989e53fe4cc9904343d81855
+    (sys.version_info[:2] == (3, 13) and sys.version_info >= (3, 13, 0, "beta", 2))
+)
```

Comment:
Instead of trying to check versions, would it be better if we just try to parse one URL to determine the version?

Reply:
> Instead of trying to check versions, would it be better if we just try to parse one URL to determine the version?

Good idea, that will save us having to update these versions. Trying that with f9e5477e68eb1c1e2c0c2256be30491705f34412